src/main/java/io/cryostat/agent/ConfigModule.java

```diff
@@ -159,6 +165,63 @@ public static Config provideConfig() {
         }
     }
 
+    @Provides
+    @Singleton
+    @Named(CRYOSTAT_AGENT_TRUSTSTORES)
+    public static List<TruststoreConfig> provideTruststoreConfigs(Config config) {
+        Map<Integer, TruststoreConfig> truststoreMap = new HashMap<>();
+
+        List<String> propertyNames =
+                StreamSupport.stream(config.getPropertyNames().spliterator(), false)
+                        .filter(e -> e.startsWith(CRYOSTAT_AGENT_TRUSTSTORES))
+                        .collect(Collectors.toList());
+
+        for (String name : propertyNames) {
+
+            String[] parts = name.split("\\.");
+            if (parts.length < 3) {
+                log.error(
+                        "Invalid truststore config property name format: {}. Rename to"
+                                + " 'truststore.cert[CERT_NUMBER].CERT_PROPERTY",
+                        name);
+            }
+
+            int truststoreNumber = 0;
+            try {
+                truststoreNumber = Integer.parseInt(parts[1].substring(5, 6));
```

Comment:
`substring(5, 6)` concerns me - this approach means there's a hardcoded cap on the number of certs that can be specified.

Maybe instead of `string.split("\\.")` it should be using regex and capture groups? This might be a little cleaner to understand. Otherwise, splitting on `.` and then doing a length-based range check in between the `[ ]` would be better than hardcoding this single-digit length.